### PR TITLE
make it possible to use directory as user source directory that doesn…

### PR DESCRIPTION
**Pull request by pcmoritz**
…'t contain worker.py

### CODE REVIEW

File: doc/using-ray-on-a-cluster.md [hunk 1]
```diff
@@ -97,41 +97,30 @@ block until the installation has completed. The standard output from the nodes
...
-the local path to the worker code that you would like to use. This command will
-copy the worker code to each node and will start the cluster. After completing
-successfully, this command will print out a command that can be run on the head
-node to attach a shell (the driver) to the cluster. For example,
+6. Start the cluster (the scheduler, object stores, and workers) with the
+command `cluster.start_ray("~/example_ray_code")`, where the argument is
+the local path to the directory that contains your Python code. This command will
+copy this source code to each node and will start the cluster. Each worker that
+is started will have a local copy of the ~/example_ray_code directory in their
+PYTHONPATH. After completing successfully, this command will print out a command
```
Comment (robertnishihara) [inline doc/using-ray-on-a-cluster.md]: it's in `sys.path`, which is not exactly the same thing as `PYTHONPATH`, right?

Comment (mehrdadn) [inline doc/using-ray-on-a-cluster.md]: Actually it is. The path variable you're thinking of is different I believe.

Comment (robertnishihara) [inline doc/using-ray-on-a-cluster.md]: Ok, maybe, but I guess when you load Python it adds a bunch of stuff.
```
> echo $PYTHONPATH
/Users/rkn/Workspace/ray/lib/python/:/Users/rkn/Workspace/ray/thirdparty/numbuf/build

> python -c "import sys; print sys.path"
['', '/Users/rkn/anaconda/lib/python2.7/site-packages/pynumbuf-0.1.dev0-py2.7.egg', '/Users/rkn/Downloads/gym', '/Users/rkn/anaconda/lib/python2.7/site-packages/six-1.10.0-py2.7.egg', '/Users/rkn/anaconda/lib/python2.7/site-packages/halo-0.1.dev0-py2.7.egg', '/Users/rkn/anaconda/lib/python2.7/site-packages/distribute-0.6.14-py2.7.egg', '/Users/rkn/Workspace/ray/lib/python', '/Users/rkn/Workspace/ray/thirdparty/numbuf/build', '/Users/rkn/anaconda/lib/python27.zip', '/Users/rkn/anaconda/lib/python2.7', '/Users/rkn/anaconda/lib/python2.7/plat-darwin', '/Users/rkn/anaconda/lib/python2.7/plat-mac', '/Users/rkn/anaconda/lib/python2.7/plat-mac/lib-scriptpackages', '/Users/rkn/anaconda/lib/python2.7/lib-tk', '/Users/rkn/anaconda/lib/python2.7/lib-old', '/Users/rkn/anaconda/lib/python2.7/lib-dynload', '/Users/rkn/.local/lib/python2.7/site-packages', '/Users/rkn/anaconda/lib/python2.7/site-packages', '/Users/rkn/anaconda/lib/python2.7/site-packages/Sphinx-1.3.5-py2.7.egg', '/Users/rkn/anaconda/lib/python2.7/site-packages/aeosa']
```

File: lib/python/ray/services.py [hunk 2]
```diff
@@ -144,7 +144,8 @@ def start_worker(worker_path, scheduler_address, objstore_address, worker_addres
   if local:
     all_processes.append((p, worker_address))
 
-def start_node(scheduler_address, node_ip_address, num_workers, worker_path=None):
```
Comment (robertnishihara) [inline lib/python/ray/services.py]: we should be consistent about breaking things into multiple lines

File: doc/using-ray-on-a-cluster.md (hunk 1 above)
Comment (mehrdadn) [inline doc/using-ray-on-a-cluster.md]: Yeah, it's the user-specified `PYTHONPATH` added to the default ones from the installation.

**Pull request merged**